src/Bicep.Core/Parsing/BaseParser.cs

```diff
@@ -805,6 +805,7 @@ private SyntaxBase LiteralValue()
                 TokenType.Integer when NumericLiteral() is { literal: Token literal, value: ulong value }
                     => new IntegerLiteralSyntax(literal, value),
                 TokenType.NullKeyword => new NullLiteralSyntax(reader.Read()),
+                TokenType.NoneKeyword => new NoneLiteralSyntax(reader.Read()),
```

Comment:
I don't think we would want `none` to behave as a literal value - it is really a special case.

For example, I don't think we would want to parse `none` in the following as a keyword:
```bicep
var test = { foo: none }
```

Reply:
I got it now, I understood the code wrong! 👍

Fixed it by removing the `NoneKeyword` case from `LiteralValue()` function